ang/civicase/activity/actions/services/tags-activity-action.service.js

```diff
@@ -3,17 +3,44 @@
 
   module.service('TagsActivityAction', TagsActivityAction);
 
+  /**
+   * Tags Activity Action Service
+   *
+   * @param {object} $rootScope rootscope object
+   * @param {object} crmApi service to use civicrm api
+   * @param {object} dialogService service to open dialog box
+   */
   function TagsActivityAction ($rootScope, crmApi, dialogService) {
+    /**
+     * Check if the Action is enabled
+     *
+     * @param {object} $scope scope object
+     * @returns {boolean} if the action is enabled
+     */
+    this.isActionEnabled = function ($scope) {
+      return $scope.mode === 'case-activity-bulk-action';
+    };
+
+    /**
+     * Perform the action
+     *
+     * @param {object} $scope scope object
+     * @param {object} action action object
+     */
+    this.doAction = function ($scope, action) {
+      manageTags(action.operation, $scope.selectedActivities, $scope.isSelectAll, $scope.params, $scope.totalCount);
```

Comment:
same comment as above.

Reply:
This will involve some time, but given that tomorrow is release and I need to send it for testing today, lets do it later.